spec/controllers/languages_controller_spec.rb

```diff
@@ -211,5 +237,69 @@
         end
       end
     end
+
+    
+    context "when logged in as an admin with policy_and_abuse role and I attempt to edit a non-abuse field" do
+      let(:admin) { create(:admin, roles: ["policy_and_abuse"]) }
+      before do
+        fake_login_admin(admin)
+        put :update, params: language_params
+      end
+      it "redirects with error" do
+        it_redirects_to_with_error(languages_path, "Policy and abuse admin can only update the 'Abuse support available' field.")
+      end
+    end 
+
+    context "when logged in as an admin with policy_and_abuse role and attempt to edit abuse_support_available field" do
+      let(:admin) { create(:admin, roles: ["policy_and_abuse"]) }
+      
+      before do
+        fake_login_admin(admin)
+        put :update, params: language_params_abuse
+      end
+      it "updates the language" do
+        finnish.reload
+        expect(finnish.name).to eq("Suomi")
+        expect(finnish.short).to eq("fi")
+        expect(finnish.support_available).to eq(false)
+        expect(finnish.abuse_support_available).to eq(false)
+        expect(finnish.sortable_name).to eq("")
+      end
+
+      it "redirects and returns success message" do
+        it_redirects_to_with_notice(languages_path, "Language was successfully updated.")
+      end
+    end 
+    
+    context "when logged in as an admin with support role and attempt to edit abuse_support_available field" do
+      let(:admin) { create(:admin, roles: ["support"]) }
+      before do
+        fake_login_admin(admin)
+        put :update, params: language_params
+      end
+      it "redirects with error" do
+        it_redirects_to_with_error(languages_path, "Support Admin cannot update the 'Abuse support available' field.")
+      end
+    end 
+
+    context "when logged in as an admin with support role and attempt to edit non-abuse fields" do
```

Comment:
This test case fails, but when I log in as a support admin on my local environment and update the support available field the change goes through, I have no idea why this is happening

Reply:
The `false` and `true` in `language_params`, `language_params_abuse` and `language_params_support` need to be `"0"` and `"1"` to match what the form sends.